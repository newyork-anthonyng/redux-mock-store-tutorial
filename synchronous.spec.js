import { addTodo } from './synchronous';

// set up redux-mock-store
import configureStore from 'redux-mock-store';
const middlewares = [];
const mockStore = configureStore(middlewares);

describe('Using redux-mock-store', () => {
  it('should add todo', () => {
    const store = mockStore({});

    store.dispatch(addTodo('Feed dog'));

    const actions = store.getActions();
    expect(actions).toMatchSnapshot();
  });
});

describe('Without redux-mock-store', () => {
  it('should add todo', () => {
    const result = addTodo('Feed dog');

    expect(result).toMatchSnapshot();
  });
});
