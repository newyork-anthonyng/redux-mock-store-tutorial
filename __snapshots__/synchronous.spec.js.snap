// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Using redux-mock-store should add todo 1`] = `
Array [
  Object {
    "todo": "Feed dog",
    "type": "ADD_TODO",
  },
]
`;

exports[`Without redux-mock-store should add todo 1`] = `
Object {
  "todo": "Feed dog",
  "type": "ADD_TODO",
}
`;
